By me: Added option to add offset from top to chars in password mode (to make centering chars possible).
---
diff --git a/dmenu.c b/dmenu.c
index 9260649..5032fd3 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -54,6 +54,7 @@ static struct item *matches, *matchend;
 static struct item *prev, *curr, *next, *sel;
 static int mon = -1, screen;
 static int lines_below_prompt = 0;
+static int password_height_offset = 0;
 
 static Atom clip, utf8;
 static Display *dpy;
@@ -157,7 +158,7 @@ drawmenu(void)
 	if (passwd) {
 	  censort = ecalloc(1, sizeof(text));
 		memset(censort, '*', strlen(text));
-		drw_text(drw, x, 0, w, bh, lrpad / 2, censort, 0);
+		drw_text(drw, x, password_height_offset*fh/100, w, bh, lrpad / 2, censort, 0);
 		curpos = TEXTW(censort) - TEXTW(&censort[cursor]);
 		free(censort);
 	} else {
@@ -879,6 +880,8 @@ main(int argc, char *argv[])
 			colors[SchemeSel][ColFg] = argv[++i];
 		else if (!strcmp(argv[i], "-w"))   /* embedding window id */
 			embed = argv[++i];
+		else if (!strcmp(argv[i], "-Po"))   /* embedding window id */
+			password_height_offset = atoi(argv[++i]);
 		else if (!strcmp(argv[i], "-d") || /* field separator */
 		         (separator_greedy = !strcmp(argv[i], "-D"))) {
 			separator = argv[++i][0];
